pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -123,7 +123,7 @@ func (c *Controller) runReconcileShootFlow(ctx context.Context, o *operation.Ope
 		})
 		_ = g.Add(flow.Task{
 			Name:         "Deploying Kubernetes API server service SNI settings in the Seed cluster",
-			Fn:           flow.TaskFn(botanist.DeployKubeAPIServerSNI).RetryUntilTimeout(defaultInterval, defaultTimeout),
+			Fn:           flow.TaskFn(botanist.Shoot.Components.ControlPlane.KubeAPIServerSNI.Deploy).RetryUntilTimeout(defaultInterval, defaultTimeout),
```

Comment:
I don't it is possible to create successfully a new Shoot after ec3ca1b9096baee324a84ce7b4bc7ae590493622. I can see that after ec3ca1b9096baee324a84ce7b4bc7ae590493622, instead of deploying the SNI resources, this step is destroying them - i.e Destroy is being called instead of Deploy. Reverting ec3ca1b9096baee324a84ce7b4bc7ae590493622 fixes the issue.

Reply:
Hm, strange 👀 Why is this? Looking at the respective commit, I don't have an explanation for such behaviour. It seems pretty straight-forward. Can you explain?

Reply:
In https://github.com/gardener/gardener/blob/v1.27.0/pkg/operation/botanist/botanist.go#L119 KubeAPIServerSNI is initially set to `b.DefaultKubeAPIServerSNI()` which returns `component.OpDestroy` component. Then in https://github.com/gardener/gardener/blob/v1.27.0/pkg/operation/botanist/controlplane.go#L816-L830 (`setAPIServerServiceClusterIP` func) KubeAPIServerSNI is overwritten to a normal component.
I guess with the new approach the flow kind of captures the reference to the default component and makes obsolete the overwrite in `setAPIServerServiceClusterIP` func.